cmd/object_put.go

```diff
@@ -0,0 +1,139 @@
+/***************************************************************
+*
+* Copyright (C) 2023, Pelican Project, Morgridge Institute for Research
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you
+* may not use this file except in compliance with the License.  You may
+* obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+***************************************************************/
+
+package main
+
+import (
+	"os"
+
+	"github.com/pelicanplatform/pelican/client"
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/pelicanplatform/pelican/param"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	putCmd = &cobra.Command{
+		Use:   "put {source ...} {destination}",
+		Short: "Send a file to a Pelican federation",
+		Run:   putMain,
+	}
+)
+
+func init() {
+	flagSet := putCmd.Flags()
+	flagSet.StringP("cache", "c", "", "Cache to use")
+	flagSet.StringP("token", "t", "", "Token file to use for transfer")
+	flagSet.BoolP("recursive", "r", false, "Recursively upload a directory.  Forces methods to only be http to get the freshest directory contents")
+	flagSet.StringP("cache-list-name", "n", "xroot", "(Deprecated) Cache list to use, currently either xroot or xroots; may be ignored")
```

Comment:
I don't think there's a concept of a cache when doing a PUT. Double check that I'm correct, and if I am, get rid of these flags. There might be a concept related to origins, so check if that's something we care about here.